### PR TITLE
Update 'ballerina/java' API docs

### PR DESCRIPTION
**Pull request by sameerajayasoma**
## Purpose
Improve API docs of ballerina/java module. 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 1]
```diff
@@ -14,29 +14,93 @@
...
+# Class is an alias for a `string` which represents a Java class name.
+public type Class string;
+
+# Identifier is an alias for a `string` that represents a Java identifier.
+public type Identifier string;
+
+# ArrayType represents a Java array type. It is used to specify parameter
+# types in `Constructor` and `Method` annotations.
+#
+# + class - element class of the array type
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + class - Element class of the array type
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 2]
```diff
@@ -14,29 +14,93 @@
...
+# + dimensions - dimensions of the array type
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + Dimensions - dimensions of the array type
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 3]
```diff
@@ -14,29 +14,93 @@
...
 |};
 
-public type MethodData record {|
-    string name?;
-    string class;
-    ParamType[] paramTypes?;
+# ConstructorData describes a Java constructor. If the `paramTypes` field is not specified,
+# then parameter types are inferred from the corresponding Ballerina function.
+#
+# + class - the class in which the constructor exists
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + class - The class in which the constructor exists
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 4]
```diff
@@ -14,29 +14,93 @@
...
+# + paramTypes - an optional field that describes parameter types of the constructor
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + paramTypes - An optional field that describes parameter types of the constructor
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 5]
```diff
@@ -14,29 +14,93 @@
...
-public type ParamType string | record {|
-    string elementClass;
-    byte dimensions;
+# MethodData describes a Java method. If the `paramTypes` field is not specified,
+# then parameter types are inferred from the corresponding Ballerina function.
+#
+# + name - an optional field that describes name of the Java method. If this field is not provided,
+#          then the name is inferred from the Ballerina function name
+# + class - the class in which the method exists
+# + paramTypes - an optional field that describes parameter types of the method
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + paramTypes - An optional field that describes parameter types of the method
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 6]
```diff
@@ -14,29 +14,93 @@
...
 
-public type ParamType string | record {|
-    string elementClass;
-    byte dimensions;
+# MethodData describes a Java method. If the `paramTypes` field is not specified,
+# then parameter types are inferred from the corresponding Ballerina function.
+#
+# + name - an optional field that describes name of the Java method. If this field is not provided,
+#          then the name is inferred from the Ballerina function name
+# + class - the class in which the method exists
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + class - The class in which the method exists
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 7]
```diff
@@ -14,29 +14,93 @@
...
+    (Class | ArrayType)[] paramTypes?;
 |};
 
-public type ParamType string | record {|
-    string elementClass;
-    byte dimensions;
+# MethodData describes a Java method. If the `paramTypes` field is not specified,
+# then parameter types are inferred from the corresponding Ballerina function.
+#
+# + name - an optional field that describes name of the Java method. If this field is not provided,
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + name - An optional field that describes name of the Java method. If this field is not provided,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 8]
```diff
@@ -14,29 +14,93 @@
...
+# + paramTypes - an optional field that describes parameter types of the method
+public type MethodData record {|
+    Identifier name?;
+    Class class;
+    (Class | ArrayType)[] paramTypes?;
 |};
 
+# FieldData describes a Java field.
+#
+# + name - an optional field that describes the name of the Java field. If this field is not provided,
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + name - An optional field that describes the name of the Java field. If this field is not provided,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 9]
```diff
@@ -14,29 +14,93 @@
...
+#          then the name is inferred from the Ballerina function name
+# + class - the class in which the field exists
```
Comment (pubudu91) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + class - The class in which the field exists
```

File: stdlib/java/src/main/ballerina/src/java/Module.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+## Module overview
+
+Provides the API for Java interoperability in Ballerina. This API includes a set of Ballerina annotations with which Java constructors, methods and fields can provide implementations to Ballerina functions with external function bodies.
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/Module.md]:
```suggestion
This API provides Java interoperability to Ballerina. It includes a set of Ballerina annotations with which Java constructors, methods, and fields can provide implementations to Ballerina functions with external function bodies.  
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 11]
```diff
@@ -14,29 +14,93 @@
 // specific language governing permissions and limitations
 // under the License.
 
-public type ConstructorData record {|
-    string class;
-    ParamType[] paramTypes?;
+# Class is an alias for a `string` which represents a Java class name.
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# Class is an alias for a `string`, which represents a Java class name.
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 12]
```diff
@@ -14,29 +14,93 @@
...
+public type Class string;
+
+# Identifier is an alias for a `string` that represents a Java identifier.
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# Identifier is an alias for a `string`, which represents a Java identifier.
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal (hunk 2 above)
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + dimensions - Dimensions of the array type
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 13]
```diff
@@ -14,29 +14,93 @@
...
+public type ArrayType record {|
+    Class class;
+    byte dimensions;
 |};
 
-public type MethodData record {|
-    string name?;
-    string class;
-    ParamType[] paramTypes?;
+# ConstructorData describes a Java constructor. If the `paramTypes` field is not specified,
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# `ConstructorData` describes a Java constructor. If the `paramTypes` field is not specified,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 14]
```diff
@@ -14,29 +14,93 @@
...
+# + paramTypes - an optional field that describes parameter types of the constructor
+public type ConstructorData record {|
+    Class class;
+    (Class | ArrayType)[] paramTypes?;
 |};
 
-public type ParamType string | record {|
-    string elementClass;
-    byte dimensions;
+# MethodData describes a Java method. If the `paramTypes` field is not specified,
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# `MethodData` describes a Java method. If the `paramTypes` field is not specified,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal (hunk 7 above)
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + name - An optional field, which describes the name of the Java method. If this field is not provided,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal (hunk 5 above)
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + paramTypes - an optional field, which describes the parameter types of the method
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 15]
```diff
@@ -14,29 +14,93 @@
...
+#          then the name is inferred from the Ballerina function name
+# + class - the class in which the method exists
+# + paramTypes - an optional field that describes parameter types of the method
+public type MethodData record {|
+    Identifier name?;
+    Class class;
+    (Class | ArrayType)[] paramTypes?;
 |};
 
+# FieldData describes a Java field.
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# `FieldData` describes a Java field.
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal (hunk 8 above)
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# + name - an optional field, which describes the name of the Java field. If this field is not provided,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 16]
```diff
@@ -14,29 +14,93 @@
...
+#          then the name is inferred from the Ballerina function name
+# + class - the class in which the field exists
 public type FieldData record {|
-    string name;
-    string class;
+    Identifier name;
+    Class class;
 |};
 
+# Constructor annotation describes a Java constructor that provides an implementation to a Ballerina function
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# The `constructor` annotation describes a Java constructor, which provides an implementation of a Ballerina function,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 17]
```diff
@@ -14,29 +14,93 @@
...
+# which has an external function body. The Ballerina function body is marked as `external` means that the
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# which has an external function body. If the Ballerina function body is marked as `external`, it means that the
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 18]
```diff
@@ -14,29 +14,93 @@
...
+# implementation of the function is not provided in the Ballerina source module.
+#
+# Following code snippet shows an example usage of this annotation. Here the `newJavaLinkedList` Ballerina function's
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# The following code snippet shows an example usage of this annotation. Here, the `newJavaLinkedList` Ballerina function's
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 19]
```diff
@@ -14,29 +14,93 @@
...
+# implementation is provided by the default constructor of `java.util.UUID.LinkedList` class.
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# implementation is provided by the default constructor of the `java.util.UUID.LinkedList` class.
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 20]
```diff
@@ -14,29 +14,93 @@
...
+#
+#       import ballerina/java;
+#
+#       function newJavaLinkedList() returns handle = @java:Constructor {
+#           class: "java.util.LinkedList"
+#       } external;
+#
 public const annotation ConstructorData Constructor on source external;
+
+# Method annotation describes a Java method that provides an implementation to a Ballerina function
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# A `method` annotation describes a Java method, which provides an implementation of a Ballerina function,
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 21]
```diff
@@ -14,29 +14,93 @@
...
+# which has an external function body. The Ballerina function body is marked as `external` means that the
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# which has an external function body. If the Ballerina function body is marked as `external`, it means that the
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 22]
```diff
@@ -14,29 +14,93 @@
...
+# implementation of the function is not provided in the Ballerina source module.
+#
+# Following code snippet shows an example usage of this annotation. Here the `getUUID` Ballerina function's
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# The following code snippet shows an example usage of this annotation. Here, the `getUUID` Ballerina function's
```

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 23]
```diff
@@ -14,29 +14,93 @@
...
+#       function getUUID() returns handle = @java:Method {
+#           name: "randomUUID",
+#           class: "java.util.UUID"
+#       } external;
+#
+# The `name` field is optional. If it is not provided, the name of the Java method is inferred
+# from the Ballerina function.
 public const annotation MethodData Method on source external;
+
+# FieldGet annotation describes a Java Field acccess that provides an implementation to a Ballerina function
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# The `FieldGet` annotation describes a Java Field access, which provides an implementation of a Ballerina function,
```
What is "a java field access"? Should it be "access to a Java field"?

File: stdlib/java/src/main/ballerina/src/java/annotations.bal [hunk 24]
```diff
@@ -14,29 +14,93 @@
...
+# which has an external function body.
 public const annotation FieldData FieldGet on source external;
+
+# FieldSet annotation describes a Java Field mutate that provides an implementation to a Ballerina function
```
Comment (praneesha) [inline stdlib/java/src/main/ballerina/src/java/annotations.bal]:
```suggestion
# The `FieldSet` annotation describes a Java Field mutate, which provides an implementation of a Ballerina function,
```

**Pull request merged**